pkg/api/agentPoolOnlyApi/v20170831/validate.go

```diff
@@ -78,6 +78,12 @@ func (a *Properties) Validate() error {
 		return e
 	}
 
+	for _, agentPoolProfile := range a.AgentPoolProfiles {
+		if e := agentPoolProfile.Validate(); e != nil {
+			return e
+		}
```

Comment:
Not blocking, but does this add missing profile validation to this API?

Reply:
this only adds agent pool length and name validation, I'll follow-up to find out about the other missing validation

Reply:
We have the validation in the service.